backend/src/helpers/databasefunctions.ts

```diff
@@ -16,31 +17,156 @@ const pool = new Pool(pgconfig);
 
 pool.on(
   'error',
-  async (_err: Error): Promise<never> => {
-    throw new ErrorHandler(400, { status: 'Database Error' });
+  async (err: Error): Promise<void> => {
+    // Uncertain how to properly solve this problem.
+    await parsePgError((err as unknown) as PgError);
   },
 );
 
+const hasConnection = async () => {
+  try {
+    await (await pool.connect()).query('SELECT NOW()');
+  } catch (e) {
+    await parsePgError(e);
+  }
+};
+
+const getClient = async (connection: Pool | Client): Promise<PoolClient | Client | undefined> => {
+  let client: any;
+  if (isPool(connection)) {
+    client = await connection.connect();
+    return client as PoolClient;
+  } else if (isClient(connection)) {
+    client = connection;
+    return client as Client;
+  } else {
+    client = undefined;
+    return client as undefined;
+  }
+};
+
+const disconnectClient = async (client: PoolClient | Client | undefined) => {
+  if (client === undefined) {
+    //TODO
+    throw new ErrorHandler(500, { type: 'Database Error' });
+  } else if (isPoolClient(client)) {
+    client.release();
+    return true;
+  } else if (isClient(client)) {
+    client.end();
+    return true;
+  }
+};
+
+const validateSQLStatement = async (sqlKeyword: string, sqlStatement: string): Promise<never | void> => {
+  const isValidSqlStatement = sqlStatement.toLowerCase().includes(sqlKeyword);
+  if (!isValidSqlStatement) {
+    throw new ErrorHandler(400, { type: 'Invalid SQL statment' });
+  }
+};
+
+const executeQuery = async (
+  sqlStatement: string,
+  connection: Pool | Client = pool,
+  data?: string[],
+): Promise<QueryResultRow | never> => {
+  let client: PoolClient | Client | undefined;
+  await hasConnection();
+  try {
+    client = await getClient(connection);
+    if (client !== undefined) {
+      const res = data ? await client.query(sqlStatement, data) : await client.query(sqlStatement);
+      return res;
+      //Consider throwing error, but it should be impossible for there not to be a connection.....
+    } else
+      throw new ErrorHandler(500, {
+        type: 'Logical Error, ask developers to review code.',
+      });
+  } catch (e) {
+    //this is only a syntactic line, the throw is only to allow typescript to not give compile errors. The throw comes from the funtion.
+    throw await parsePgError(e, sqlStatement);
+  } finally {
+    try {
+      await disconnectClient(client);
+    } catch (e) {
+      throw e;
+    }
+  }
+};
+
+export const executeTransaction = async (
```

Comment:
Right now, no. 

But it will be used when we create new companies or companyusers, because if either making the role or inserting the role into the correct table fails neither should be done.